src/Microsoft.Azure.WebJobs.Host/JobHostConfiguration.cs

```diff
@@ -123,6 +125,31 @@ public string HostId
             }
         }
 
+        /// <summary>
+        /// Gets the host machine ID. All host instances on the same machine will have the same HostMachineId.
+        /// </summary>
+        public string HostMachineId
+        {
+            get
+            {
+                if (_hostMachineId == null)
+                {
+                    _hostMachineId = Environment.GetEnvironmentVariable("WEBSITE_INSTANCE_ID")
+                        ?? Environment.MachineName.GetHashCode().ToString("X", CultureInfo.InvariantCulture).PadLeft(32, '0');
```

Comment:
>GetHashCode [](start = 51, length = 11)

I believe GetHashCode() is not deterministic and could return different results when called in different process invocations or different appdomains.